pkg/controller/common/keystore/initcontainer.go

```diff
@@ -68,20 +68,26 @@ func initContainer(
 		return corev1.Container{}, err
 	}
 
+	volumeMounts := []corev1.VolumeMount{
+		// access secure settings
+		secureSettingsSecret.VolumeMount(),
+	}
+
+	// caller might be already taking care of the right mount and volume
+	if parameters.DataVolumePath != "" {
+		// volume mount to write the keystore in the data volume
+		volumeMounts = append(volumeMounts, DataVolume(volumePrefix, parameters.DataVolumePath).VolumeMount())
```

Comment:
I am -1 on adding additional conditional logic here that is just used for one controller. Either we always rely on `WithInitContainerDefaults` to copy the data volume mount into the keystone init container, then we should just remove this parameter. Or we do it consistently for all invocations via the parameter. 

Let me know if I am missing something here ofc.

Reply:
I've reverted that change. I renamed the data volume to match the volume name in the init container. If that feels off we can follow up separately, as this is what we are already doing in ES and I'd like to avoid expanding scope of this PR.